src/lib/profiles/data-management/Current/TraitData.cpp

```diff
@@ -91,16 +91,16 @@ WEAVE_ERROR UpdateDictionaryDirtyPathCut::CutPath(PropertyPathHandle aPathhandle
 WEAVE_ERROR TraitSchemaEngine::ParseTagString(const char * apTagString, char ** apEndptr, uint8_t & aParseRes) const
 {
     WEAVE_ERROR err = WEAVE_NO_ERROR;
-
+    unsigned long int tag;
     VerifyOrExit(apTagString != NULL, err = WEAVE_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(*apTagString == '/', err = WEAVE_ERROR_INVALID_ARGUMENT);
 
     apTagString++;
 
-    aParseRes = strtoul(apTagString, apEndptr, 0);
+    tag = strtoul(apTagString, apEndptr, 0);
     VerifyOrExit(!(*apEndptr == apTagString || (**apEndptr != '\0' && **apEndptr != '/')), err = WEAVE_ERROR_INVALID_ARGUMENT);
```

Comment:
This code fails to check for ERANGE errors, e.g. when the value given exceeds that which will fit within unsigned long.  See ParseInt() in nlargparser for required logic: https://github.com/openweave/openweave-core/blob/7627fe106fa56abcecfc126de1c21b0f24fe39be/src/lib/support/nlargparser.cpp#L698-L708

Reply:
In principle, you're absolutely correct.  In practice, the runtime difference in the code is which  error we return -- the `ERANGE` case is effectively handled by the check in line 102.  I believe either error is appropriate to return in the `ERANGE` case; the main difference is that if at some point the `kContextTagMaxNum` became larger than `ULONG_MAX` the code would no longer work (although in that case it should give us the compile time warning that precipitated this change).

I typed in the above paragraphs largely because I was wary of checking in device code that relied on `errno`.  Only at the end did I process that this code is executing on platforms that need to process strings, etc. 

I will change this to follow specs of `strtoul`.

Reply:
I tried the fixes with `errno` in place, ended up running into issues with embedded targets still somehow pulling in this method.  Given the reasoning above, I'm going to go back to the original code.